### PR TITLE
#136 Variants list command 

### LINKED ISSUE #136: Command to list all available Variants
## Background

- At the moment, the only way to know which variants are available is to check the `variants.yml` file, which may be more difficult to read and harder to access when working in the Terminal

## Solution

We could implement a command `variants list` that could work similar to `git branch` in listing all the available Variants in a project based on what's listed in the spec file:
```
  default
* dev
  qa
```

### PR DESCRIPTION
**Pull request by obackbase**
### What does this PR do
New command `$ variants list` introduced, that shows all available variants

### How can it be tested
Tell us how to test these changes

### Task
resolves #136 

### Checklist:

- [x] I ran `make validation` locally with success
- [x] I have not introduced new bugs
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new errors

### CODE REVIEW

**Comment by gderaco** [comment]
@obackbase  I run the command and there's a lot of space between the executed command and the output

File: Sources/VariantsCore/Commands/List.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
+//
+//  Variants
+//
+//  Copyright (c) Backbase B.V. - https://www.backbase.com
+//  Created by Oleg Baidalka on 31.03.2022.
```
Comment (gderaco) [inline Sources/VariantsCore/Commands/List.swift:5]: please use the same headers as on the other files

File: Sources/VariantsCore/Schemas/Variant.swift [hunk 2]
```diff
@@ -0,0 +1,12 @@
+//
+//  Variants
+//
+//  Copyright (c) Backbase B.V. - https://www.backbase.com
+//  Created by Oleg Baidalka on 31.03.2022.
```
Comment (gderaco) [inline Sources/VariantsCore/Schemas/Variant.swift:5]: please use the same headers as on the other files

File: Tests/VariantsCoreTests/CommandTests.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
+//
+//  Variants
+//
+//  Copyright (c) Backbase B.V. - https://www.backbase.com
+//  Created by Oleg Baidalka on 29.04.2022.
```
Comment (gderaco) [inline Tests/VariantsCoreTests/CommandTests.swift:5]: please use the same headers as on the other files

**Review by gderaco — CHANGES_REQUESTED** [review]

File: Sources/VariantsCore/Commands/List.swift (hunk 1 above)
Comment (obackbase) [reply Sources/VariantsCore/Commands/List.swift:5]: like what? It is like other files

**Comment by obackbase** [comment]
> @obackbase I run the command and there's a lot of space between the executed command and the output

this was made with the purpose to keep the space for the selected variant. Once `isCurrent(Variant)` function implemented correctly it will print with `* ` to highlight current variant (same as in `git branch`)

File: Sources/VariantsCore/Commands/List.swift (hunk 1 above)
Comment (gderaco) [reply Sources/VariantsCore/Commands/List.swift:5]: the date should not be there

Comment (obackbase) [reply Sources/VariantsCore/Commands/List.swift:5]: ah. ok

**Review by gderaco — APPROVED** [review]
LGTM

File: Sources/VariantsCore/Commands/List.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        let detectedPlatform = try PlatformDetector.detect(fromArgument: platform)
+        let project = ProjectFactory.from(platform: detectedPlatform, logger: logger)
+        let variants = try project.list(spec: spec)
+
+        let output = Logger(verbose: true)
+        variants.forEach { output.log($0.print(project: project)) }
+    }
+}
+
+extension Variant {
```
Comment (GMinucci) [inline Sources/VariantsCore/Commands/List.swift:46]: Should this be a `fileprivate` extension? If not then can we move it to a separate file?

Comment (obackbase) [reply Sources/VariantsCore/Commands/List.swift:46]: was before `fileprivate` but can't tests are firing errors, so I remove that
> VerboseLoggerTests.swift:57:32: 'print' is inaccessible due to 'fileprivate' protection level

**Review by GMinucci — APPROVED** [review]

**Pull request merged**